kpi-parent/kpi-budget-chart/src/main/java/objective/taskboard/followup/budget/BudgetChartService.java

```diff
@@ -59,16 +59,12 @@ private BudgetChartData filterByTimeline(BudgetChartData data, DashboardTimeline
         if (data.isEmpty())
             return data;
 
-        Optional<LocalDate> optionalStartDate = timeline.startDate;
-        Optional<LocalDate> optionalEndDate = timeline.endDate;
-
-        if (!optionalStartDate.isPresent() || !optionalEndDate.isPresent())
-            return data;
-        
-        LocalDate startDate = optionalStartDate.get();
-        LocalDate endDate = optionalEndDate.get();
-        
-        return filterByRange(data, RangeUtils.between(startDate, endDate));    
+        return timeline
+                .hasBothDates((startDate, endDate) -> {
+                    Range<LocalDate> dateRange = RangeUtils.between(startDate, endDate);
+                    return filterByRange(data, dateRange);
+                })
+                .orElse(data);
```

Comment:
It's simpler to understand
```suggestion
        if (!timeline.hasBothDates())
            return data;

        Range<LocalDate> dateRange = RangeUtils.between(timeline.startDate.get(), timeline.endDate.get());
        return filterByRange(data, dateRange);
```